acvm_backend_barretenberg/src/composer.rs

```diff
@@ -165,14 +163,15 @@ impl Composer for Barretenberg {
 
     fn verify_with_vk(
         &self,
+        reference_string: &CRS,
         constraint_system: &ConstraintSystem,
         // XXX: Important: This assumes that the proof does not have the public inputs pre-pended to it
         // This is not the case, if you take the proof directly from Barretenberg
         proof: &[u8],
         public_inputs: Assignments,
         verification_key: &[u8],
     ) -> bool {
-        let g2_data = G2::new().data;
+        let CRS { g2_data, .. } = reference_string;
```

Comment:
This change is good as it moves away from the G2 struct. One unified CRS is easier to reason about